crates/re_ui/src/ui_ext.rs

```diff
@@ -13,6 +13,65 @@ use crate::{
 
 static FULL_SPAN_TAG: &str = "rerun_full_span";
 
+/// Icon button to be used in the header of a panel.
+pub struct HeaderMenuButton<'a> {
+    pub icon: &'static Icon,
+    pub add_contents: Box<dyn FnOnce(&mut egui::Ui) + 'a>,
+    pub enabled: bool,
+    pub hover_text: Option<String>,
+    pub disabled_hover_text: Option<String>,
+}
+
+impl<'a> HeaderMenuButton<'a> {
+    pub fn new(icon: &'static Icon, add_contents: impl FnOnce(&mut egui::Ui) + 'a) -> Self {
+        Self {
+            icon,
+            add_contents: Box::new(add_contents),
+            enabled: true,
+            hover_text: None,
+            disabled_hover_text: None,
+        }
+    }
+
+    /// Sets enable/disable state of the button.
+    #[inline]
+    pub fn with_enabled(mut self, enabled: bool) -> Self {
```

Comment:
I think we should do without the `with_` prefix. (ikr, "do as I say, not as I do")

Reply:
Hmm isn't that how we do our builder pattern elsewhere, e.g. in re_types?

Reply:
If anything, we're extremely inconsistent. ListItem2 itself has 80% its builder patter function without with. 99% of egui doesn't use `with_` either. Id say this latter point is motivation enough to avoid it within `ui_ext.rs` at least :)

Reply:
👍